src/component/segment_time.rs

```diff
@@ -0,0 +1,214 @@
+            key_abbreviations: Box::new(["Segment Time".into(), "Seg. Time".into()]) as _,
+            display_two_rows: self.settings.display_two_rows,
+        }
+    }
+
+    /// Accesses a generic description of the settings available for this
+    /// component and their current values.
+    pub fn settings_description(&self) -> SettingsDescription {
+        SettingsDescription::with_fields(vec![
+            Field::new("Background".into(), self.settings.background.into()),
+            Field::new(
+                "Comparison".into(),
+                self.settings.comparison_override.clone().into(),
+            ),
+            Field::new(
+                "Display 2 Rows".into(),
+                self.settings.display_two_rows.into(),
+            ),
+            Field::new("Label Color".into(), self.settings.label_color.into()),
+            Field::new("Value Color".into(), self.settings.value_color.into()),
+            Field::new("Accuracy".into(), self.settings.accuracy.into()),
+        ])
+    }
+
+    /// Sets a setting's value by its index to the given value.
+    ///
+    /// # Panics
+    ///
+    /// This panics if the type of the value to be set is not compatible with
+    /// the type of the setting's value. A panic can also occur if the index of
+    /// the setting provided is out of bounds.
+    pub fn set_value(&mut self, index: usize, value: Value) {
+        match index {
+            0 => self.settings.background = value.into(),
+            1 => self.settings.comparison_override = value.into(),
+            2 => self.settings.display_two_rows = value.into(),
+            3 => self.settings.label_color = value.into(),
+            4 => self.settings.value_color = value.into(),
+            5 => self.settings.accuracy = value.into(),
+            _ => panic!("Unsupported Setting Index"),
+        }
+    }
+}
+
+// TODO: Move this into the analysis module. "calculate_segment_time" or
+// something. (copied from the detailed timer)
+fn calculate_comparison_time(
+    timer: &Timer,
+    timing_method: TimingMethod,
+    comparison: &str,
+    last_split_index: usize,
+) -> Option<TimeSpan> {
+    let run = timer.run();
+
+    if comparison == best_segments::NAME {
+        run.segment(last_split_index).best_segment_time()[timing_method]
+
+    } else if last_split_index == 0 {
+        run.segment(0).comparison_timing_method(comparison, timing_method)
+
+    } else if timer.current_split_index() > Some(0) {
+        let last_split_index = if last_split_index >= timer.run().len() {
+            run.len() - 1
+        } else {
+            last_split_index
+        };
+
+        Some(
+            run
+                .segment(last_split_index)
+                .comparison_timing_method(comparison, timing_method)?
+                - run
+                    .segment(last_split_index - 1)
+                    .comparison_timing_method(comparison, timing_method)?,
+        )
+    } else {
+        None
+    }
+}
```

Comment:
There's some stuff I'm still unsure about here:
- If this should go under `crate::analysis::state_helper`
- The fix for split indexes larger than `run.len() - 1` causing panics. (lines 197-201)
- Code formatting.

Reply:
> If this should go under `crate::analysis::state_helper`

That does in fact seem to be the best place for it, but the name you suggested conflicts with another function in the detailed timer.

> If the fix for split indexes larger than `run.len() - 1` causing panics is acceptable. (lines 197-201)

It wasn't even working properly, but I fixed that in my latest commit.

> General code formatting.

Honestly I don't think there is much I can do in that regard.